src/main/java/org/openrewrite/csharp/dependencies/DependencyInsight.java

```diff
@@ -15,16 +15,37 @@
  */
 package org.openrewrite.csharp.dependencies;
 
-import org.openrewrite.ExecutionContext;
-import org.openrewrite.Recipe;
-import org.openrewrite.TreeVisitor;
+import lombok.EqualsAndHashCode;
+import lombok.Value;
+import org.jspecify.annotations.Nullable;
+import org.openrewrite.*;
 import org.openrewrite.csharp.dependencies.trait.PackageReference;
+import org.openrewrite.internal.StringUtils;
 import org.openrewrite.marker.SearchResult;
 import org.openrewrite.maven.table.DependenciesInUse;
+import org.openrewrite.semver.Semver;
 
+@Value
+@EqualsAndHashCode(callSuper = false)
 public class DependencyInsight extends Recipe {
     transient DependenciesInUse dependenciesInUse = new DependenciesInUse(this);
 
+    @Option(displayName = "Artifact pattern",
+            description = "Artifact ID glob pattern used to match dependencies.",
+            example = "Microsoft*",
+            required = false)
+    @Nullable
+    String artifactIdPattern;
```

Comment:
In .NET / NuGet land people talk about packages rather than artifacts, so we may want to rename things and also adjust display names and descriptions accordingly, as to not confuse the user.

Reply:
Would you even go as far as from `DependencyInsight` to `PackageInsight` ?

Reply:
Not sure. I think dependency might still be a generic enough term. While the C# project files use the term package references, I think developers probably also refer to them as dependencies. But I am guessing a bit here.

Reply:
Then let's stick with just the renamed field for now then, to keep symmetry with the Maven/Gradle and Nodejs recipes and visualizations we already have. We can gather feedback around this from the first users.